app/plugin_key_value_store.go

```diff
@@ -59,3 +59,13 @@ func (a *App) DeletePluginKey(pluginId string, key string) *model.AppError {
 
 	return result.Err
 }
+
+func (a *App) ListPluginKeys(pluginId string) ([]string, *model.AppError) {
+	result := <-a.Srv.Store.Plugin().List(pluginId)
+
+	if result.Err != nil {
+		mlog.Error(result.Err.Error())
+	}
+
+	return nil, result.Err
```

Comment:
It appears that this doesn't actually return the plugin keys from `result`: just `nil`. Can you update the unit tests for `TestPluginKeyValueStore` to exercise this new functionality?

Reply:
@lieut-data Thanks for reviewing the code. 
By mistake I removed the condition where I was checking for result.Data != nil and returning keys. 

I have another question though. When we insert the keys, they key names are hashed. I get the hashed value back from the db but there is no way to convert it back to original string. Should I be returning hashed value back?

Reply:
Huh, I had no idea that was the implementation until now. The only reason to do that would be to allow arbitrarily large keys as input, but that really ought to be a client-side decision (i.e. the plugin) given the risk of collisions and the inability to recover the original keys.

I'm going to raise this one on https://pre-release.mattermost.com/core/channels/developer-toolkit to discuss further.

Reply:
Will wait for a decision before updating the PR.

Reply:
Added test.